src/lib.rs

```diff
@@ -564,39 +559,36 @@ impl StreamDeck {
     /// they will appear on the device!
     pub fn clear_button_image(&self, key: u8) -> Result<(), StreamDeckError> {
         self.initialize()?;
-        match self.kind {
-            Kind::Akp153 | Kind::Akp153E | Kind::Akp153R | Kind::Akp815 | Kind::Akp03E | Kind::Akp03R | Kind::MiraBoxHSV293S => {
-                let key = if self.kind == Kind::Akp815 {
-                    inverse_key_index(&self.kind, key)
-                } else if self.kind == Kind::Akp03E || self.kind == Kind::Akp03R {
-                    key
-                } else {
-                    elgato_to_ajazz153(&self.kind, key)
-                };
 
-                let mut buf = vec![0x00, 0x43, 0x52, 0x54, 0x00, 0x00, 0x43, 0x4c, 0x45, 0x00, 0x00, 0x00, if key == 0xff { 0xff } else { key + 1 }];
+        if self.kind.is_mirabox() {
+            let key = match self.kind {
+                Kind::Akp815 => inverse_key_index(&self.kind, key),
+                Kind::Akp03E | Kind::Akp03R => key,
+                _ => elgato_to_ajazz153(&self.kind, key),
+            };
 
-                ajazz_extend_packet(&self.kind, &mut buf);
+            let mut buf = vec![0x00, 0x43, 0x52, 0x54, 0x00, 0x00, 0x43, 0x4c, 0x45, 0x00, 0x00, 0x00, if key == 0xff { 0xff } else { key + 1 }];
 
-                write_data(&self.device, buf.as_slice())?;
+            ajazz_extend_packet(&self.kind, &mut buf);
 
-                Ok(())
-            }
+            write_data(&self.device, buf.as_slice())?;
 
-            _ => Ok(self.send_image(key, &self.kind.blank_image())?),
+            return Ok(());
         }
+
```

Comment:
I think an `else` block without using the `return` keyword is more Rust-y.

Reply:
Fixed